### PR TITLE
Ripped out old dependencies and allow php 8

### PR DESCRIPTION
**Pull request by chives**

### CODE REVIEW

File: lib/FSi/DoctrineExtensions/Translatable/Mapping/Driver/Annotation.php [hunk 1]
```diff
@@ -52,7 +54,8 @@ protected function loadExtendedClassMetadata(
                 continue;
             }
 
-            $translatableAnnotation = $this->getAnnotationReader()->getPropertyAnnotation($property, self::TRANSLATABLE);
+            $translatableAnnotation = $this->getAnnotationReader()
```
Comment (szymach) [inline lib/FSi/DoctrineExtensions/Translatable/Mapping/Driver/Annotation.php:57]:
```php
$translatableAnnotation = $this->getAnnotationReader()->getPropertyAnnotation(
    $property, 
    self::TRANSLATABLE
);
```

File: lib/FSi/DoctrineExtensions/Uploadable/Keymaker/TransliterateEntity.php [hunk 2]
```diff
@@ -52,9 +53,6 @@ private function cleanFileName(string $originalName): string
             $originalName
         );
 
-        return trim(preg_replace(
-            '/[-\s]+/', '-',
-            preg_replace('/[^\da-z.\-\s]+/', '', $transliteratedName)
-        ));
+        return trim(preg_replace('/[-\s]+/', '-', preg_replace('/[^\da-z.\-\s]+/', '', $transliteratedName)));
```
Comment (szymach) [inline lib/FSi/DoctrineExtensions/Uploadable/Keymaker/TransliterateEntity.php:58]: You could add uses for these functions as well.

File: tests/FSi/DoctrineExtensions/Tests/Tool/BaseORMTest.php [hunk 3]
```diff
@@ -174,16 +129,15 @@ protected function getEntityManager(): EntityManagerInterface
...
-            $evm
-        );
+        $conn = DriverManager::getConnection(['driver' => 'pdo_sqlite', 'memory' => true], $config, $evm);
         $em = EntityManager::create($conn, $config, $evm);
 
-        $schema = array_map(function($class) use ($em) {
-            return $em->getClassMetadata($class);
-        }, (array) $this->getUsedEntityFixtures());
+        $schema = array_map(
+            static function ($class) use ($em) {
```
Comment (szymach) [inline tests/FSi/DoctrineExtensions/Tests/Tool/BaseORMTest.php]: Typehints?

File: tests/FSi/DoctrineExtensions/Tests/Translatable/BaseTranslatableTest.php [hunk 4]
```diff
@@ -39,24 +39,22 @@ abstract class BaseTranslatableTest extends BaseORMTest
...
-    ) {
+        string $title = self::POLISH_TITLE_1,
+        string $subtitle = self::POLISH_SUBTITLE,
+        string $contents = self::POLISH_CONTENTS_1,
+        ?string $locale = null
+    ): Article {
         $article = new Article();
         $article->setDate(new DateTime());
-        $article->setLocale($locale ? $locale : self::LANGUAGE_PL);
+        $article->setLocale($locale ?: self::LANGUAGE_PL);
```
Comment (szymach) [inline tests/FSi/DoctrineExtensions/Tests/Translatable/BaseTranslatableTest.php]: `$locale ?? self::LANGUAGE_PL`

**Comment by chives** [comment]
@fsi-open

**Review by szymach — APPROVED** [review]

**Pull request merged**